src/util/track.rs

```diff
@@ -0,0 +1,150 @@
+//! Track memory ranges for tools like Valgrind address sanitizer, or other memory checkers.
+
+use crate::util::Address;
+
+pub fn tracking_enabled() -> bool {
+    #[cfg(feature = "crabgrind")]
+    {
+        crabgrind::run_mode() != crabgrind::RunMode::Native
+    }
+
+    #[cfg(not(feature = "crabgrind"))]
+    {
+        false
+    }
+}
+
+pub fn track_malloc(p: Address, size: usize, zero: bool) {
+    #[cfg(feature = "crabgrind")]
+    {
+        crabgrind::memcheck::alloc::malloc(p.to_mut_ptr(), size, 0, zero)
+    }
+
+    #[cfg(not(feature = "crabgrind"))]
+    {
+        let _ = p;
+        let _ = size;
+        let _ = zero;
+    }
+}
+
+pub fn track_free(p: Address, size: usize) {
```

Comment:
`size` can be removed. `crabgrind::memcheck::alloc::free` asks for `rz` (https://docs.rs/crabgrind/latest/crabgrind/memcheck/alloc/fn.free.html). Is it redzone size, rather than allocation size?